### DISCUSSION TITLE
"Go to definition" for certain compiler headers links to clangd, even when compiling with GCC

### DISCUSSION THREAD

**Discussion by recursivenomad** [clangd/clangd#2130] [Q&A] [answered]
I am using clangd as my open-source LSP, but compiling with GCC.  I almost have my set-up playing nice together, and most compiler headers resolve to the compiler I am using (in this case, arm-none-eabi-gcc).  But when I follow certain system header includes (for example `#include <stdbool.h>`), it directs me to the clangd implementation of said header, rather than to the compiler called by compile_commands.json (which is the same compiler approved via `--query-driver`).

I am seeking advice on how to get clangd to resolve all system headers to the compiler being used (or at least not fall back to clangd's implementation if it is unable to locate said header) 🙏

**Comment by HighCommander4** [comment, marked as answer]
This behaviour changed recently in clangd 18. See https://github.com/clangd/clangd/issues/1695 for some background and discussion.

If you prefer using gcc's built-in headers, the current workaround is to add the path containing them to the compile command manually, using an `-isystem` flag in https://clangd.llvm.org/config.html#compileflags. https://github.com/clangd/clangd/issues/2074 is on file to provide an easier way of doing this.

**Reply by recursivenomad** [reply]
Thank you as always, Nathan 🙏
